src/Npgsql/NpgsqlWriteBuffer.cs

```diff
@@ -280,7 +280,9 @@ public void WriteDouble(double value, bool littleEndian)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         void Write<T>(T value)
         {
-            Debug.Assert(Unsafe.SizeOf<T>() <= WriteSpaceLeft);
+            if (Unsafe.SizeOf<T>() > WriteSpaceLeft)
+                throw new InvalidOperationException("There is not enough space left in the buffer.");
+            
             Unsafe.WriteUnaligned(ref Buffer[WritePosition], value);
             WritePosition += Unsafe.SizeOf<T>();
```

Comment:
Is this second call elided? Or should we store the first call in a local?

Reply:
[This is](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il) how `Unsafe` is implemented.